app/models/mark.rb

```diff
@@ -18,13 +18,29 @@ class Mark < ApplicationRecord
   validates_uniqueness_of :markable_id,
                           scope: [:result_id, :markable_type]
 
+  validates_inclusion_of :override, in: [true, false]
+
+  def calculate_deduction
+    return 0 if self.markable_type != 'FlexibleCriterion' || !self.override?
+    total_deduction = 0
+    applied_annotations = mark.result.annotations
+    applied_annotations.each do |annotation|
```

Comment:
This is good, but you can speed it up with a smarter database query. https://guides.rubyonrails.org/active_record_querying.html#calculations.

Reply:
I actually have this wrong I think, I did a sum after your comment, but now I realize that this will add all the deductions to the mark, even the ones for other criteria if there are.